content/blog/kubecon-2019/index.md

```diff
@@ -0,0 +1,272 @@
+
+So the question we ask you is: are you about to break prod? We will all break production at some point because there's things we miss. As Alexandra points out in her tweet, "failures are part of the cost of building and shipping large systems". Building a robust pipeline, allows us to move quickly in the case of failure and gain confidence around making changes to our infrastructure and applications.
+
+{{< tweet 1198373475049623552 >}}
+
+With this in mind, we will use Pulumi and GitLab to build a pipeline that validates both our application, infrastructure, and deployment process.
+
+## Ephemeral Environments
+
+What is an ephemeral environment? It is a short-lived environment that mimics our production environment. To maintain agility, we typically define the boundaries of this environment to only encompass the first-level dependencies of the particular microservice we are deploying. This means, we will not spin up every single microservice or piece of infrastructure thats running in production. Yet we may need to spin up some extra pieces infrastructure to be able to properly test our microservice. For example, we may need to create a subscription to pull from a PubSub topic your microservice writes to.
+
+## Why this is important
+
+Infrastructure is often a key part of our application's behavior. Our architecture and requirements are constantly evolving. How can we incorporate these into our testing suite to give us a high degree of confidence?
+
+Ephemeral environments allow us to integrate our infrastructure and deployment process into our testing suite. They ensure our testing environment is always in-sync with production and therefore allows us to iterate quickly to meet new requirements.
+
+Ephemeral environments also encourage us to lean on automated tests over manual tests. If we use ephemeral environments as a replacement for a testing environment, there is not enough time to go in and run a manual check. Shifting our mindsets to use automated tests can be challenging, yet its imperative that we do so. Automated tests guarantee our application behaves as expected today as well as months from now when we're out on vacation.
+
+## Our Demo Application
+
+To demonstrate the effectiveness of integrating acceptance testing with ephemeral environments into your deployment process, we created a simple demo application. The service is written in Go and accepts a message on the `/message` endpoint, then places it in a storage bucket and sends a notification about the new object on a PubSub topic. The code for this application lives in our [main.go](https://gitlab.com/rocore/demo-app/blob/master/main.go) file. While you can walk through this code yourself, the most important thing to call out is that our application is configurable. This means we take configuration in at the very beginning of our main function and shut down the application if the values are not present.
```

Comment:
Maybe bold or italicize configurable just to make it stand out because I think it one of the super key points in our
presentation. Just a suggestion.

"the most important thing to call out is that our application is **_configurable_**"